pyadjoint/checkpointing.py

```diff
@@ -182,6 +204,9 @@ def _(self, cp_action, timestep):
                 # Activate disk checkpointing only in the checkpointing process.
                 for package in self.tape._package_data.values():
                     package.pause_checkpointing()
+
+            if self._gc_timestep_frequency and timestep % self._gc_timestep_frequency == 0:
+                gc.collect(self._gc_generation)
```

Comment:
I would be tempted to add a `logging.info("Running a garbage collection cycle")` or similar here. Logging is already used inside pyadjoint [here](https://github.com/dolfin-adjoint/pyadjoint/blob/da58d45c5a4dafe529e215f977e108fbabc2f360/pyadjoint/control.py#L50) for example

Reply:
Done.